src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -0,0 +1,535 @@
+            CourseAttributes course, FeedbackSessionAttributes session, InstructorAttributes instructor,
+            String template, String subject)
+                    throws MessagingException, IOException {
+        
+        String submitUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                                 .withCourseId(course.getId())
+                                 .withSessionName(session.getFeedbackSessionName())
+                                 .toAbsoluteString();
+        
+        String reportUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_FEEDBACK_RESULTS_PAGE)
+                                 .withCourseId(course.getId())
+                                 .withSessionName(session.getFeedbackSessionName())
+                                 .toAbsoluteString();
+        
+        String emailBody = Templates.populateTemplate(template,
+                "${userName}", instructor.name,
+                "${courseName}", course.getName(),
+                "${courseId}", course.getId(),
+                "${feedbackSessionName}", session.getFeedbackSessionName(),
+                "${deadline}", TimeHelper.formatTime12H(session.getEndTime()),
+                "${instructorFragment}", "",
+                "${submitUrl}", submitUrl,
+                "${reportUrl}", reportUrl);
+        
+        MimeMessage email = getEmptyEmailAddressedToEmail(instructor.email);
+        email.setSubject(String.format(subject, course.getName(), session.getFeedbackSessionName()));
+        email.setContent(emailBody, "text/html");
+        return email;
+    }
+    
+    /**
+     * Generates the feedback session closing emails for the given {@code session}.
+     */
+    public List<MimeMessage> generateFeedbackSessionClosingEmails(FeedbackSessionAttributes session)
+            throws MessagingException, IOException {
+        
+        List<StudentAttributes> students = new ArrayList<StudentAttributes>();
+        
+        if (fsLogic.isFeedbackSessionForStudentsToAnswer(session)) {
+            List<StudentAttributes> studentsForCourse = studentsLogic.getStudentsForCourse(session.getCourseId());
+            
+            for (StudentAttributes student : studentsForCourse) {
+                try {
+                    if (!fsLogic.isFeedbackSessionFullyCompletedByStudent(session.getFeedbackSessionName(),
+                            session.getCourseId(), student.email)) {
+                        students.add(student);
+                    }
+                } catch (EntityDoesNotExistException e) {
+                    log.severe("Course " + session.getCourseId() + " does not exist or "
+                               + "session " + session.getFeedbackSessionName() + " does not exist");
+                    // Course or session cannot be found for one student => it will be the case for all students
+                    // Do not waste time looping through all students
+                    break;
```

Comment:
@damithc, actually, separate from this issue, do we have statistics about how the cron jobs for emails are performing on the live server? e.g. does it ever fail? are we in danger of the cron jobs failing due to performance issues?